packages/dds/tree/src/feature-libraries/optional-field/optionalFieldCodecV1.ts

```diff
@@ -0,0 +1,129 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TAnySchema, Type } from "@sinclair/typebox";
+import { IJsonCodec } from "../../codec/index.js";
+import {
+	ChangeAtomId,
+	ChangeEncodingContext,
+	EncodedRevisionTag,
+	RevisionTag,
+} from "../../core/index.js";
+import { JsonCompatibleReadOnly, Mutable } from "../../util/index.js";
+import { makeChangeAtomIdCodec } from "../changeAtomIdCodec.js";
+import type { EncodedChangeAtomId, NodeChangeset } from "../modular-schema/index.js";
+import { EncodedOptionalChangeset, EncodedRegisterId } from "./optionalFieldChangeFormatV1.js";
+import type { OptionalChangeset, RegisterId, Replace } from "./optionalFieldChangeTypes.js";
+
+function makeRegisterIdCodec(
```

Comment:
IMO rather than duplicate, import from V0 here and in the format file. Newer versions depending on older versions (because they haven't changed anything) seems fine.

Reply:
I was wondering about that. In this case they are slightly different (one take the changeAtomIdCodec while the other takes the revisionTagCodec). I could still make them the same by making them both take the changeAtomIdCodec though. I just wasn't sure whether it was worth the code savings given that it muddies the water a bit: introducing a new format should not change the old one. This wouldn't change the old format (just the codec logic for it) but it does make the PR noisier. All in all I don't have a strong opinion, so if you'd rather I reduce the duplication I'm happy to do it.

Reply:
I just didn't scrutinize enough :). Since they're different, I think it's fine.